beacon-chain/state/types.go

```diff
@@ -190,7 +190,7 @@ func newValHandler(vals []*ethpb.Validator) *validatorMapHandler {
 }
 
 // copies the whole map and returns a map handler with the copied map.
-func (v *validatorMapHandler) copy() *validatorMapHandler {
+func (v *validatorMapHandler) copyHandler() *validatorMapHandler {
```

Comment:
I was hesitating whether diagnostics should be reported for `copy` because it's a common function naming pattern. I eventually decided that name conflicts will only happen for non-exported functions, for which proper naming is a little bit less important.

And `copy` is a Go built-in function after all.

Reply:
+1 , i dont really see any benefit to changing common method names/patterns here. This just adds in an extra layer of verbosity when trying to understand the method.

Reply:
This will be reverted. I did not take into consideration that functions with receiver should be treated differently because you cannot access such actions directly. And therefore they do not shadow anything.